### DISCUSSION TITLE
Agenda not working

### DISCUSSION THREAD

**Discussion by pabloat81** [TypedProject/ts-express-decorators#1520] [General]
Hi. I am trying the Example of agenda implementation of the doc Page. 

I Installed agenda and tsed agenda,. 

at server.ts added the agenda opt with enable. 

Then created a class copied from doc and in the method with @Every tried to send an email, console.log, save something in db but never executes.

On server start y read on thr Last line: add agenda definitions... Only.

Do i have to do something else? The agenda class must be in some particular folder?

Thanks in advance.

**Comment by Romakita** [comment]
@ochrstn

**Comment by ochrstn** [comment]
> 
> Do i have to do something else? The agenda class must be in some particular folder?
> 
> Thanks in advance.

I guess you also have to import the file with the `@Agenda`-decorated-class somewhere (e.g. Server.ts) so it can get picked up as a Provider

**Reply by pabloat81** [reply]
Hi. I can try. 

Docs dont mention that.

I tried putting the class un protocols folder. Didnt work.

Thanks.

**Comment by ochrstn** [comment]
I just created an example repository here that shows a minimal setup: https://github.com/ochrstn/tsed-agenda-example

**Comment by pabloat81** [comment]
Tried everything and didnt work, till i added:
db: {
      address: defaultConfig.url
    }

and worked... why is nessesary that config?

Other question: can i call services from agenda method?

Thanks

**Reply by Romakita** [reply]
Yes you can inject any services in any class decorated with @Agenda decorator.
